lib/api/objectCopy.js

```diff
@@ -400,10 +405,12 @@ function objectCopy(authInfo, request, sourceBucket,
             return async.mapLimit(dataLocator, 1,
                 // eslint-disable-next-line prefer-arrow-callback
                 function copyPart(part, cb) {
-                    return data.get(part, null, log, (err, stream) => {
+                    const passThrough = new PassThrough();
+                    return data.get(part, passThrough, log, (err, stream) => {
                         if (err) {
                             return cb(err);
                         }
+                        const objStream = stream || passThrough;
```

Comment:
I would have azure pass back a stream in its callback so you don't need this special case.